kaldb/src/main/java/com/slack/kaldb/preprocessor/PreprocessorService.java

```diff
@@ -195,8 +194,9 @@ protected static Topology buildTopology(
     ValueMapper<byte[], Iterable<Trace.Span>> valueMapper =
         PreprocessorValueMapper.byteArrayToTraceSpans(dataTransformer);
 
+    int lingerMs = Integer.parseInt(kafkaProperties.getProperty("producer.linger.ms", "100"));
     StreamPartitioner<String, Trace.Span> streamPartitioner =
-        streamPartitioner(datasetMetadataList);
+        new PreprocessorPartitioner<>(datasetMetadataList, lingerMs);
```

Comment:
Let's pass in all the kafka properties here so we can use other props in the future.

Reply:
addressed in the latest change